### DISCUSSION TITLE
Autopilot disconnection to early

### DISCUSSION THREAD

**Discussion by CptBayer** [flybywiresim/aircraft#5251] [General]
Hi dev team,

the last 3 legs i had always autopilot discos in cruise level! Aircraft was shaking a bit due to windy weather. On each leg the wind wasn´t that strong. In my opinion the disconnections are to sensitive .... 

Greets Klaus

**Comment by aguther** [comment]
Hello,

about which version of the plane are you talking?

The development version does not disconnect due to turbulence yet. Only if:
- high speed protection
- high angle of attack protection
- too much side stick (> 50%)
- too much rudder (> 40%)

**Reply by CptBayer** [reply]
Hi,

sorry, i am using the dev version. I experienced the folling things not:

- side stick more than 50 % -didn´t touch due to inflight with AP on
- rudder more than 40 % - didn´t use due to inflight with AP on
- high speed protection - only light wind changes with no overspeed situation
- high angle of attack - as well not. Afterr the last asobo update I have a lot off freezes. After the freeze is over the nose is going very deep down, without AP disco !

but the light bumpy weather forced the discos ... hmmm .... 

greets klaus

**Reply by aguther** [reply]
I wrote the code and tell you it’s one of those, not the turbulence, here is the line of code:
https://github.com/flybywiresim/a32nx/blob/master/src/fbw/src/FlyByWireInterface.cpp#L482

Beside that of course also the AP disconnect button will disconnect the AP.

Double check if you have any assignments you’re not aware of and also ensure there is no unintentional sidestick or rudder input.

We had several cases where users initially thought it’s a bug and in the end…we had even users with strange hardware issues…

**Reply by CptBayer** [reply]
Hi Aguther,

I am using the TCA Thrustmaster with all addional stuff (flaps + slat lever), as well the sidestick of TCA. All the axis are set as they should (sensitivity, dead zone etc.). I have only a few assignments on the buttons which I didn´t press during or short of the discos... (for your info, I am on board with flight simulation since round about 30 years, so I my knowledge base is not so bad)... The code is nice, but I am not a programmer :-p ..... I can´t check the code for any kind of errors :-p .... 

Last but not least I have the discos since a short time and didn´t experineced them before. Nevertheless I will observe the next legs and report if the issue will come again. I am 100 % sure that I  hadn´t any kind of unintentional input in any axis ...

Greets Klaus

**Reply by aguther** [reply]
Ok, I use also the TCA Quadrant, but not the sidestick. We had Someone on discord observing a disconnect, that happened when we reset the camera. That’s what he said:
> Hi A! I have investigated this a bit more. I have no double binding, but it is the hatswitch on my TCA side stick that disconnects it. Super strange. Only AP control I have is the red button on the stick.

So maybe it’s an MSFS issue or whatever. There was no change since a while on the AP and I just don’t see anything that could trigger it beside the bespoke ones.

I cannot rule out a bug 100%, but it’s very very unlikely.

Feel free to provide screenshots / videos or an FDR file of a flight where it happened. Location of the FDR files is described here:
https://docs.flybywiresim.com/start/autopilot-fbw/#reporting-issues

**Reply by CptBayer** [reply]
Hi aguther,


interresting aspect with the hatswitch. I also have side views on the hatswitch. In generalI I am using a headtracker and did not use the hatswitch at all. 

On my leg today, bumby weather, AP disco without touching anything. Speedtape was nearly alphafloor. I didn´t see the change in speed. Was to fast. After accelerating with nose done I activated the AP again. Short after this I had a overspeed warning with no significant wind changes.... Groovy.... I don´t made a film or somethin else.... IF it happend next time I will do some screenshots..

Gruß Klaus

**Reply by AdamNZ** [reply]
My Dev version has been working well right up until around a week ago. I now get A/T disconnects, waypoints out of sequence and A/P malfunctions on almost every flight. This seems to happen when nearing the first STAR waypoint.

**Reply by CptBayer** [reply]
Hi,

just for info, I had today also the next AP disco .... spooky ...

Greets klaus

**Reply by aguther** [reply]
Yeah this discussion is useless without any further data. Please send me the FDR files of the flight(s) in question and I can take a look and maybe find out what was going on.

**Reply by CptBayer** [reply]
ok. After my holidays I´ll send you an fdr file ..

Greets Klaus
